weave-js/src/components/PagePanelComponents/PublishModal.tsx

```diff
@@ -47,6 +48,29 @@ const iconStyle = {
   marginRight: 8,
 };
 
+const useLastVisitedPath = () => {
+  const [lastVisitedPath] = useLocalStorage<string>('lastVisited', '');
+  const [lastVisitedEntity, lastVisitedProject] = useMemo(() => {
+    let e = null;
+    let p = null;
+    if (lastVisitedPath !== '') {
+      const crumbs = lastVisitedPath.split('/').slice(1);
+      if (
+        crumbs.length >= 3 &&
+        crumbs[0] === 'browse' &&
+        crumbs[1] === 'wandb'
+      ) {
+        e = crumbs[2];
+        if (crumbs.length >= 4) {
+          p = crumbs[3];
+        }
+      }
+    }
+    return [e, p];
+  }, [lastVisitedPath]);
+  return [lastVisitedEntity, lastVisitedProject];
```

Comment:
Might be a little more self-documenting if instead of a tuple your return value was an object like: 
```
{
    "entityName": "",
    "projectName": "",
}
```

Reply:
changed in next commit